### ISSUE TITLE
ISigner

### ISSUE THREAD

**Issue by shargon** [CityOfZion/neo-sharp#540]
- Create a ISigner<ConsensusMessage> for https://github.com/shargon/neo-sharp/blob/development/src/NeoSharp.Core/Messaging/Messages/ConsensusMessage.cs

Questions

- Should we unify this ISigner into only one? (a lot of classes for the same, compute a hash)
- Should we rename ISigner for something more... intuitive (is not a sign)?

**Comment by osmirnov** [comment]
Yes, ISigner naming and design can get a bit more care. If we can unify logic in a single place it is even better.
